tmt/steps/prepare/install.py

```diff
@@ -10,6 +10,253 @@
+        # Parse the copr repo name
+        matched = re.match("^(@)?([^/]+)/([^/]+)$", copr)
+        if not matched:
+            raise tmt.utils.PrepareError(f"Invalid copr repository '{copr}'.")
+        group, name, project = matched.groups()
+        group = 'group_' if group else ''
+        # Prepare the repo file url
+        parts = [COPR_URL] + (['g'] if group else [])
+        parts += [name, project, 'repo', 'epel-6']
+        parts += [f"{group}{name}-{project}-epel-6.repo"]
+        url = '/'.join(parts)
+        # Download the repo file on guest
+        try:
+            guest.execute(f'curl -LOf {url}', cwd='/etc/yum.repos.d')
+        except tmt.utils.RunError as error:
+            if 'not found' in error.stderr.lower():
+                raise tmt.utils.PrepareError(
+                    f"Copr repository '{copr}' not found.")
+            raise
+
+    def enable_copr(self, command, plugin, guest):
+        """ Enable requested copr repositories """
+        coprs = self.data.get('copr')
+        if not coprs:
+            return
+        # Try to install copr plugin
+        self.debug('Make sure the copr plugin is available.')
+        try:
+            guest.execute(f'rpm -q {plugin} || {command} install -y {plugin}')
+        # Enable repositories manually for epel6
+        except tmt.utils.RunError:
+            for copr in coprs:
+                self.info('copr', copr, 'green')
+                self.enable_copr_epel6(copr, guest)
+        # Enable repositories using copr plugin
+        else:
+            for copr in coprs:
+                self.info('copr', copr, 'green')
+                guest.execute(f'{command} copr enable -y {copr}')
+
+    def prepare_install_local(self, step):
+        """ Copy packages to the test system """
+        self.rpms_directory = os.path.join(step.workdir, 'rpms')
+        os.makedirs(self.rpms_directory)
+
+        # Copy local packages into workdir, push to guests
+        for package in self.local_packages:
+            self.verbose(os.path.basename(package), shift=1)
+            self.debug(f"Copy '{package}' to '{self.rpms_directory}'.", level=3)
+            shutil.copy(package, self.rpms_directory)
+        for guest in step.plan.provision.guests():
+            guest.push()
+
+    def install_debuginfo(self):
+        """ Install debuginfo packages """
+        packages = self.prepare_packages(
+            self.debuginfo_packages, title="debuginfo")
+        # Make sure debuginfo-install is present on the target system
+        self.guest.execute(f"{self.command} install -y /usr/bin/debuginfo-install")
+        self.guest.execute(f"debuginfo-install -y {packages}")
+
+    def install(self, step):
+        """ Perform the actual installation """
+        if len(self.local_packages) > 0:
+            self.prepare_install_local(step)
+            self.install_local()
+        if len(self.repo_packages) > 0:
+            self.install_from_repository()
```

Comment:
`install_from_repository()` is not a method of `InstallBase`, I'd expect the static analysis to report this issue since it cannot be proven this method would exist in runtime. It should be added, either as an abstract method or with a default no-op implementation.